File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> merged;
        if(intervals.size()==0) return merged;
        sort(intervals.begin(),intervals.end());
        vector<int> tempInterval=intervals[0];
        for(auto it:intervals){
            if(it[0]<=tempInterval[1]){
                tempInterval[1]=max(it[1],tempInterval[1]);
            }else{
                merged.push_back(tempInterval);
                tempInterval=it;
            }
        }
        merged.push_back(tempInterval);
        return merged;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    ''' Brute O(NlogN)+O(2N)
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        n=len(intervals)
        intervals.sort()
        i=0
        ans=[]
        while i<n:
            start=intervals[i][0]
            end=intervals[i][1]
            j=i+1
            while j<n and intervals[j][0]<=end:
                end=max(end,intervals[j][1])
                j+=1
            ans.append([start,end])
            i=j
        return ans
    '''
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        ans=[]
        intervals.sort()
        for int in intervals:
            if not ans or ans[-1][1]<int[0]:
                ans.append(int)
            else:
                ans[-1][1]=max(ans[-1][1],int[1])
        return ans